### ISSUE TITLE
[2022 CVPR] Long-Tailed Recognition via Weight Balancing

### ISSUE THREAD

**Issue by Jasonlee1995** [Jasonlee1995/AI_Papers#81]
Long-tailed recognition task에서 기존의 `Decoupling Representation and Classifier for Long-Tailed Recognition` 논문에서 제안한 2-stage training 방법을 이용 + weight decay를 튜닝하여 sota를 찍었다는 논문
(즉 weight decay를 잘 tuning하라는 논문)

논문에서 핵심만 간단히 요약

### 1.1. Preliminaries

Long-tailed train dataset은 imbalance factor인 $\textbf{IF} = \frac{\textrm{max}_k \begin{vmatrix} D_k \end{vmatrix}}{\textrm{min}_k \begin{vmatrix} D_k \end{vmatrix}} \gg 1$ 을 만족해야함
( $D_k$ 는 k-class에 해당하는 data의 개수)

Balanced dataset에서의 accuracy를 통해 evaluation

Reference 논문인 `Decoupling Representation and Classifier for Long-Tailed Recognition`를 알아야 논문이 이해가 됨
(간단하게만 설명하면 representation은 잘 학습하는데 classification의 weight가 dataset과 같이 imbalance되어있으며, 이를 post-hoc 방법을 이용하여 개선할 수 있다... 정도)

저자들은 논문에서는 weight balancing technique 관점에서 classification의 weight를 balancing 하고자 함

### 1.2. Weight Balancing Technique
<img width="935" alt="image" src="https://user-images.githubusercontent.com/49643709/183303299-3d5847eb-5f55-4c65-b282-3aed6d7f654f.png">

#### 1.2.1. L2-normalization
<img width="471" alt="image" src="https://user-images.githubusercontent.com/49643709/183302095-cc31fc56-d877-4354-922b-e84deff06a28.png">

Classifier의 weight가 무조건 unit-length를 가지도록 constraint를 두어 학습

위의 방식처럼 학습하지 않고, post-hoc 방식으로 L2-normalize, $\tau$ -normalize하는 방식도 존재

Post-hoc 방식으로 했을 때의 overall performance가 낮아지는 문제점이 발견

저자들은 이에 대해 L2-normalization은 long-tailed distributed classes를 balancing하는데 너무 strict하다고 표현

Constraint를 두고 학습하면 naive training보다는 괜찮지만 weight decay, MaxNorm보다 underperform

#### 1.2.2. Weight Decay
<img width="463" alt="image" src="https://user-images.githubusercontent.com/49643709/183302832-f0f3b85f-e6a2-421e-a154-4bd5f9f38c68.png">

Weight decay는 small balanced weights를 학습하도록 encourage함
(penalize more heavily on large weights)

기존의 LTR task에서 weight decay에 대한 별다른 연구 등이 진행되지 않았는데, 단순히 weight decay만 잘 튜닝해도 sota를 찍을 수 있음

#### 1.2.3. MaxNorm Constraint
<img width="457" alt="image" src="https://user-images.githubusercontent.com/49643709/183303095-92e1e7c3-7743-4fab-a691-30571eb747e5.png">

L2-normalization은 weight가 unit-length를 가지게 제약을 걸었는 반면, MaxNorm은 length의 maximum에 제약을 검
(weight norm이 radius가 $\delta$인 ball 안에 들어가도록 제약)

L2-normalization이 norm value를 1이 되도록 strict하게 제약을 거는 반면 MaxNorm은 이러한 제약을 relax

### 1.3. Training Pipeline

2-stage training

1. Feature learning : train a network using the standard cross-entropy loss by properly tuning weight decay
2. Classifier learning : train a classifier using a class-balanced loss by tuning weight decay and MaxNorm with freezing the learned features

### 2.1. Ablation Study
<img width="471" alt="image" src="https://user-images.githubusercontent.com/49643709/183400128-3df41111-d9b8-431f-81bc-9814c04ba6a1.png">

#### 2.1.1. Weight Decay
<img width="472" alt="image" src="https://user-images.githubusercontent.com/49643709/183393607-36f88ed7-f93c-4a0d-9657-55df4a214269.png">

Weight decay를 제외한 다른 hyperparameter는 fix한 채로 first-stage training만 진행

Optimal weight decay는 dataset마다 다르며, weight decay만 충분히 tuning해도 accuracy가 증가함

#### 2.1.2. How to Regularize Classifier Learning
<img width="937" alt="image" src="https://user-images.githubusercontent.com/49643709/183394543-b5c794e8-edf1-42d6-8622-e96931584e6f.png">

1. 저자들은 첫번째로 classifier가 balanced weights를 학습해야함을 보임
    > 학습을 하지 않아도 되는 post-hoc 방식인 $\tau$-normalization을 classifier의 weight를 balancing을 통해 성능이 42%에서 51.31%까지 오름

2. 두번째로 class-balanced (CB) loss를 이용하여 classifier를 학습하는 것이 crucial함을 보임
    > Regularization 없이 CB loss만 이용하여 학습하면 성능이 46.08%에서 47.09% 까지 밖에 오르고, weight decay를 넣고 학습하면 성능이 52.42%까지 오름

3. MaxNorm을 같이 사용하면 더 좋고 더 많은 layer를 학습하면 더 좋음을 보임
    > MaxNorm과 weight decay를 같이 적용하면 53.35%까지, non-linear MLP classifier를 사용하면 53.55%까지 성능 증가

### 2.2. Benchmark Results
<img width="453" alt="image" src="https://user-images.githubusercontent.com/49643709/183399942-045f2cc8-84b7-408b-970d-7e288811d0fd.png">
<img width="429" alt="image" src="https://user-images.githubusercontent.com/49643709/183400022-251d18d4-5133-4633-ba55-6d0c1fbaae05.png">
